### PR TITLE
Changes added for attachments to salesforce sobjects.

### PR DESCRIPTION
**Pull request by vikasrathee-cs**
Changes added for attachments to salesforce sobjects.

### CODE REVIEW

File: src/main/java/io/cdap/plugin/salesforce/SalesforceBulkUtil.java [hunk 1]
```diff
@@ -62,12 +62,12 @@ public final class SalesforceBulkUtil {
    */
   public static JobInfo createJob(BulkConnection bulkConnection, String sObject, OperationEnum operationEnum,
                                   @Nullable String externalIdField,
-                                  ConcurrencyMode concurrencyMode) throws AsyncApiException {
+                                  ConcurrencyMode concurrencyMode, ContentType contentType) throws AsyncApiException {
```
Comment (anup-codex) [inline src/main/java/io/cdap/plugin/salesforce/SalesforceBulkUtil.java:72]: Please add some docs stating which `ContentType` are currently being used.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceRecordWriter.java [hunk 2]
```diff
@@ -69,33 +80,65 @@ public SalesforceRecordWriter(TaskAttemptContext taskAttemptContext) throws IOEx
...
       submitCurrentBatch();
     }
 
+    if (isFileUploadObject) {
+      populateInputStreamMap(record);
+    }
     csvBuffer.write(csvRecord);
   }
 
+  private int getFileUploadSize(StructuredRecord record) {
```
Comment (anup-codex) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceRecordWriter.java:113]: Better to have java doc stating the size limitation.

**Review by anup-codex — COMMENTED** [review]
Looks good, minor comments about the documentation and add some unit tests.

File: src/main/java/io/cdap/plugin/salesforce/SalesforceBulkUtil.java (hunk 1 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/SalesforceBulkUtil.java:72]: done.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceRecordWriter.java (hunk 2 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceRecordWriter.java:113]: done.

Comment (anup-codex) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceRecordWriter.java:113]: add space in-between `bytes.Exception`.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java [hunk 3]
```diff
@@ -340,6 +340,11 @@ private void validateSchema(Schema schema, FailureCollector collector, @Nullable
         break;
       case upsert:
         externalIdFieldName = getExternalIdField();
+        if (externalIdFieldName == null) {
```
Comment (anup-codex) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java]: should check for empty/blank also.

**Review by anup-codex — APPROVED** [review]

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java (hunk 3 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java]: done

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceRecordWriter.java (hunk 2 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceRecordWriter.java:113]: done

**Pull request merged**